component/azstorage/azstorage.go

```diff
@@ -352,10 +352,15 @@ func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (length
 }
 
 func (az *AzStorage) WriteFile(options internal.WriteFileOptions) (int, error) {
-	err := az.storage.WriteFromBuffer(options.Handle.Path, nil, options.Data)
+	err := az.storage.Write(options.Handle.Path, options.Offset, int64(len(options.Data)), options.Data, options.FileOffsets, options.ModBlockList)
```

Comment:
Like we did for our component interface, shall we also pass a structure instead of so many arguments. That will make it easy to modify in the future.

Reply:
I can make that change in a separate pr to change all the azstorage methods to take in options bags, I don't think this change should be made here.